docs/aws/audit/dynamodbmonitoring/rules/ddb_table_backup_enabled.mdx

```diff
@@ -24,6 +24,208 @@ HIPAA, NIST, SOC2, ISO27001, HITRUST, AWSWAF, NISTCSF
+3. In the navigation pane on the left, click on 'Tables'.
+4. Select the table you want to check for continuous backup. 
+5. In the 'Overview' tab, scroll down to the 'Backup' section. Here, you can see if 'Point-in-time recovery' is enabled or not. If it is enabled, it means continuous backup is enabled for the table.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   Installation:
+   ```
+   pip install awscli
+   ```
+   Configuration:
+   ```
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all DynamoDB tables: Use the AWS CLI command `aws dynamodb list-tables` to get a list of all DynamoDB tables in your account. The command is as follows:
+   ```
+   aws dynamodb list-tables --region your-region-name
+   ```
+   Replace 'your-region-name' with the name of your AWS region.
+
+3. Describe each table: For each table in the list, use the `aws dynamodb describe-continuous-backups` command to get the backup details. Replace 'your-table-name' with the name of your DynamoDB table.
+   ```
+   aws dynamodb describe-continuous-backups --table-name your-table-name --region your-region-name
+   ```
+4. Check the backup status: In the output of the above command, look for the `ContinuousBackupsDescription` field. If the `ContinuousBackupsStatus` is 'ENABLED', then continuous backup is enabled for the table. If not, then it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Create a DynamoDB resource object using the session:
+
+```python
+dynamodb = session.resource('dynamodb')
+```
+
+4. Iterate over all the tables in your DynamoDB and check if continuous backups are enabled:
+
+```python
+for table in dynamodb.tables.all():
+    backup = dynamodb.meta.client.describe_continuous_backups(TableName=table.name)
+    if 'ContinuousBackupsDescription' in backup:
+        if backup['ContinuousBackupsDescription']['ContinuousBackupsStatus'] == 'ENABLED':
+            print(f"Continuous backup is enabled for {table.name}")
+        else:
+            print(f"Continuous backup is NOT enabled for {table.name}")
+    else:
+        print(f"Continuous backup is NOT enabled for {table.name}")
+```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>